### PR TITLE
Add ingest-attachment-with-arrays section to ingest attachments doc

### PR DESCRIPTION
**Pull request by gameldar**
Added a new section detailing how to use the attachment processor
within an array.

This reverts commit #22296 and instead links to the foreach processor.

Note this invalidates the previous pull request (#22295) which was made against the wrong branch.

### CODE REVIEW

**Comment by gameldar** [comment]
Note this was for Issue #22294

File: docs/reference/ingest/ingest-node.asciidoc [hunk 1]
```diff
@@ -1166,6 +1166,10 @@ In this example, if the `remove` processor does fail, then
 the array elements that have been processed thus far will
 be updated.
 
+Another advanced example can be found in {plugins}/using-ingest-attachment.html[attachment processor documentation].
```
Comment (dadoonet) [inline docs/reference/ingest/ingest-node.asciidoc]: Can you change the link to `{plugins}/ingest-attachment-with-arrays.html` so people are redirected to right page automatically?

**Review by dadoonet — CHANGES_REQUESTED** [review]
Left a small comment but we are getting close now! :)

**Review by dadoonet — APPROVED** [review]

**Comment by dadoonet** [comment]
Ok so I tried your branch and ran `gradle check` in docs dir and it failed.
Well, I forgot about this. You need to add `// NOTCONSOLE` after each snippet or `// CONSOLE` if you want it to be executed as a test.

The later is better as we are sure that any time we update the project, the doc stills accurate.

You have a great example in `analysis-icu.asciidoc` if you wish.

Thanks!

**Review by dadoonet — CHANGES_REQUESTED** [review]
Ok so I tried your branch and ran `gradle check` in docs dir and it failed.
Well, I forgot about this. You need to add `// NOTCONSOLE` after each snippet or `// CONSOLE` if you want it to be executed as a test.

The later is better as we are sure that any time we update the project, the doc stills accurate.

You have a great example in `analysis-icu.asciidoc` if you wish.

Thanks!

**Review by dadoonet — APPROVED** [review]
I tested it locally and it's great!
Thanks a lot! 
Going to merge it now.

**Comment by dadoonet** [comment]
Backported:

* 5.x: da6fb27
* 5.1: e1c6cf2

**Pull request merged**